### PR TITLE
[AMD] Supporting fused kernels build using JIT

### PR DESCRIPTION
**Pull request by R0n12**
This PR aims to enable JIT compilation for fused kernels on AMD GPUs, so that the same code will run on both AMD and NVIDIA GPUs. 

Previous `python setup.py install` method has issues with hipifying files, which results in ninja generating multiple rules for the same target. 

This will also make the extra fused kernel building instruction in the readme unnecessary. I will take care of the readme parts once this get approved

- [x] builds on AMD (MI100, MI250X) and NVIDIA (A100)
- [x] `test_fused_kernels.py` passed

#954 
Reference: https://github.com/microsoft/Megatron-DeepSpeed/commit/b4d4a0e06577218a21abf18eb7d8088bb8ecaad3

### CODE REVIEW

File: megatron/fused_kernels/__init__.py [hunk 1]
```diff
@@ -12,13 +12,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Copyright (c) 2022, NVIDIA CORPORATION. All rights reserved.
```
Comment (Quentin-Anthony) [inline megatron/fused_kernels/__init__.py:15]: Why include this copyright?

Comment (StellaAthena) [reply megatron/fused_kernels/__init__.py:15]: Unless this file is verbatim from NVIDIA's code, the copyread header should look more like [this](https://github.com/EleutherAI/gpt-neox/blob/9ddff09d360fa9e281c96df48ad1bf42c05eb5f1/megatron/training.py)

Comment (R0n12) [reply megatron/fused_kernels/__init__.py]: This code is borrowed from 
https://github.com/microsoft/Megatron-DeepSpeed/blob/main/megatron/fused_kernels/__init__.py

I have pushed a correct format

Comment (StellaAthena) [reply megatron/fused_kernels/__init__.py]: It's not quite the same wording as we use elsewhere, but the new header looks adequate.

**Comment by StellaAthena** [comment]
@Quentin-Anthony should we update the README to reflect this change?

**Comment by Quentin-Anthony** [comment]
> @Quentin-Anthony should we update the README to reflect this change?

Yes. We're on it now.

**Pull request merged**